### PR TITLE
refactor(resourcegroups): use resourcegroupstaggingapi

### PR DESCRIPTION
**Pull request by iamhopaul123**
Use resourcegroupstaggingapi instead of use resourcegroups api for `GetResourcesByTags` method. After this change `GetResourcesByTags` will also return all tags for the found resources.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/aws/cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ import (
 
 const (
 	resourceQueryType      = "TAG_FILTERS_1_0"
```
Comment (efekarakus) [inline internal/pkg/aws/cloudwatch/cloudwatch.go:21]: Is this used now?

File: internal/pkg/aws/resourcegroups/resourcegroups.go [hunk 2]
```diff
@@ -32,69 +27,63 @@ type tagFilter struct {
...
-	resourceResp := &resourcegroups.SearchResourcesOutput{}
-
-	query, err := rg.searchResourcesQuery(resourceType, tags)
-	if err != nil {
-		return nil, fmt.Errorf("construct search resource query: %w", err)
+// GetResourcesByTags gets tag set and ARN for the resource with input resource type and tags.
+func (rg *ResourceGroups) GetResourcesByTags(resourceType string, tags map[string]string) ([]*Resource, error) {
+	var resources []*Resource
+	var err error
+	resourceResp := &resourcegroupstaggingapi.GetResourcesOutput{}
```
Comment (efekarakus) [inline internal/pkg/aws/resourcegroups/resourcegroups.go:47]: nit: define after first for-loop

File: internal/pkg/aws/resourcegroups/resourcegroups.go [hunk 3]
```diff
@@ -32,69 +27,63 @@ type tagFilter struct {
...
-	var resourceArns []string
-	resourceResp := &resourcegroups.SearchResourcesOutput{}
-
-	query, err := rg.searchResourcesQuery(resourceType, tags)
-	if err != nil {
-		return nil, fmt.Errorf("construct search resource query: %w", err)
+// GetResourcesByTags gets tag set and ARN for the resource with input resource type and tags.
+func (rg *ResourceGroups) GetResourcesByTags(resourceType string, tags map[string]string) ([]*Resource, error) {
+	var resources []*Resource
+	var err error
```
Comment (efekarakus) [inline internal/pkg/aws/resourcegroups/resourcegroups.go:46]: do we need this variable? can we initialize it inside the for-loop instead

File: internal/pkg/aws/resourcegroups/resourcegroups.go [hunk 4]
```diff
@@ -32,69 +27,63 @@ type tagFilter struct {
 	Values []string
 }
 
-type query struct {
-	ResourceTypeFilters []string
-	TagFilters          []tagFilter
+// Resource contains the ARN and the tags of the resource.
+type Resource struct {
+	Arn  string
```
Comment (efekarakus) [inline internal/pkg/aws/resourcegroups/resourcegroups.go:32]: nit: 
```suggestion
	ARN  string
```
https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: internal/pkg/aws/resourcegroups/resourcegroups.go [hunk 5]
```diff
@@ -32,69 +27,63 @@ type tagFilter struct {
...
-		for _, identifier := range resourceResp.ResourceIdentifiers {
-			arn := aws.StringValue(identifier.ResourceArn)
-			resourceArns = append(resourceArns, arn)
+		for _, resourceTagMapping := range resourceResp.ResourceTagMappingList {
+			tags := make(map[string]string)
+			for _, tag := range resourceTagMapping.Tags {
+				if tag.Key == nil {
+					continue
+				}
+				tags[*tag.Key] = aws.StringValue(tag.Value)
```
Comment (efekarakus) [inline internal/pkg/aws/resourcegroups/resourcegroups.go:70]: isnt it possible that the value is nil? maybe:
```go
tags[*tag.Key] = ""
if tag.Value != nil {
  tags[*tag.Key] = aws.StringValue(tag.Value)
}
```

File: internal/pkg/aws/resourcegroups/resourcegroups.go [hunk 6]
```diff
@@ -32,69 +27,63 @@ type tagFilter struct {
...
 		}
-		if resourceResp.NextToken == nil {
+		// usually pagination token is "" when it doesn't have any next page. However, since it
+		// is type *string, it is safer for us to check nil value for it as well.
+		if resourceResp.PaginationToken == nil {
 			break
+		} else {
+			if aws.StringValue(resourceResp.PaginationToken) == "" {
+				break
+			}
```
Comment (efekarakus) [inline internal/pkg/aws/resourcegroups/resourcegroups.go:84]:
```suggestion
		if tok := resourceResp.PaginationToken; tok == nil || aws.StringValue(tok) == ""{
			break
		} 
```

File: internal/pkg/aws/resourcegroups/resourcegroups.go (hunk 5 above)
Comment (iamhopaul123) [reply internal/pkg/aws/resourcegroups/resourcegroups.go:70]: "StringValue returns the value of the string pointer passed in or if the pointer is nil."

Comment (efekarakus) [reply internal/pkg/aws/resourcegroups/resourcegroups.go:70]: lol that's good to know

**Pull request merged**